### ISSUE TITLE
createDirective type reduced in .d.ts

### ISSUE THREAD

**Issue by FlatMapIO** [vobyjs/voby#25]
The generic parameter `T extends keyof JSX.Directives` has collapsed to `T extends never` in the `.d.ts` file.

```tsx
declare const createDirective: <T extends never>(name: T, fn: DirectiveFunction<JSX.Directives[T]>, options?: DirectiveOptions) => Directive<T>;
```

I think this is an issue with the TypeScript compiler. Perhaps we need to find a way to work around it and keep the normal generic parameters.

**Comment by FlatMapIO** [comment]
I searched TypeScript Issues and found that this problem has been scheduled for TypeScript 5.1 🤣
https://github.com/microsoft/TypeScript/issues/49536

**Comment by fabiospampinato** [comment]
It should be fixed in v0.48.1, I added two nonsense builtin directives to the interface to make TS not fall under the buggy path.

Thanks for the report 👍
